eduNEXT/eox-tenant
# feat: add support for JWT authentication `DS-959`

## Description
This PR allows you to make a request with a JWT token.

## Testing instructions
Using **Tutor**:
1. Install the plugin with these changes. If you are installing it for the first time you should run the migrations.
2. Create a new application from `http://local.edly.io:8000/admin/oauth2_provider/application/`. Associate a user with sufficient permissions and add to the **Redirect uris** field: http://local.edly.io:8000/
3. Generate a JWT Token for the user using the next endpoint. **POST** `http://local.edly.io:8000/oauth2/access_token/`. The **content type** of the request must be `application/x-www-form-urlencoded`

    **Body parameters**
    
    - `client_id`: Client ID of the application created above. e.g. **CAfZv...IHD0J**
    - `client_secret`: Client Secret of the application created above. e.g. **Qu2pnJJ9b1C6DA...u7Ej8x31RFiaKg9**
    - `grant_type`: **client_credentials**
    - `token_type`: **JWT**

    **Response**
    ```json
      {
	      "access_token": "eyJhbGciOiJIUz...SsDAs3KE_gEc",
	      "expires_in": 3600,
	      "token_type": "JWT",
	      "scope": "read write email profile"
      }
    ```

    - `access_token`: The user's access token. You must use this token in the Authorization header of the requests to the API.
4. Test all `eox-tenant` endpoints using the JWT token, all should work correctly. You can use this [eox-tenant](https://github.com/eduNEXT/eox-tenant/files/15505801/eox-tenant.postman_collection.json) postman collection.

## Checklist for Merge
